core/server/src/main/java/alluxio/worker/block/BlockWorkerClientRestServiceHandler.java

```diff
@@ -63,14 +66,17 @@
   public static final String UNLOCK_BLOCK = "unlock_block";
   public static final String WRITE_BLOCK = "write_block";
 
-  private final BlockWorker mBlockWorker = AlluxioWorkerService.Factory.get().getBlockWorker();
-  private final StorageTierAssoc mStorageTierAssoc = new WorkerStorageTierAssoc();
+  private final BlockWorker mBlockWorker;
+  private final StorageTierAssoc mStorageTierAssoc;
 
   /**
-   * Constructs a new {@link BlockWorkerClientRestServiceHandler}.
+   * @param context context for the servlet
    */
-  public BlockWorkerClientRestServiceHandler() {}
-
+  public BlockWorkerClientRestServiceHandler(@Context ServletContext context) {
+    mBlockWorker = ((AlluxioWorkerService) context
+        .getAttribute(WorkerUIWebServer.ALLUXIO_WORKER_SERVLET_RESOURCE_KEY)).getBlockWorker();
+    mStorageTierAssoc = new WorkerStorageTierAssoc();
```

Comment:
I think `WorkerStorageTierAssoc`s should probably be owned by `BlockWorker`s. Then we could get it out of the passed-in block worker. This would require some refactoring though and isn't necessary for this PR